libsolidity/ast/Types.cpp

```diff
@@ -940,7 +940,7 @@ BoolResult RationalNumberType::isExplicitlyConvertibleTo(Type const& _convertTo)
 	if (category == Category::FixedBytes)
 		return false;
 	else if (category == Category::Address)
-		return !(isNegative() || isFractional() || integerType()->numBits() > 160);
+		return !(isNegative() || isFractional() || !integerType() || integerType()->numBits() > 160);
 	else if (category == Category::Integer)
 		return false;
 	else if (auto enumType = dynamic_cast<EnumType const*>(&_convertTo))
```

Comment:
The enum check uses `m_value` directly, while for address we go through the integer conversion helper. I think we should do the same for enums? This would find any disagreements between what is happening on conversion vs. a direct comparison of rational.